/*
 * This file is part of rasdaman community.
 *
 * Rasdaman community is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Rasdaman community is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with rasdaman community.  If not, see <http://www.gnu.org/licenses/>.
 *
 * Copyright 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013,
 *2014 Peter Baumann / rasdaman GmbH.
 *
 * For more information please see <http://www.rasdaman.org>
 * or contact Peter Baumann via <baumann@rasdaman.com>.
 */

// This package contains messages and services used for testing.
package rnp_test;
option java_package = "org.rasdaman.rasnet.message";

// This option is mandatory for creating service stubs
option cc_generic_services = true;
option java_generic_services = true;

// An empty message that can be sent as a request
message DummyRequest { optional int32 requestNo = 1; }

// An empty message that can be sent as a reply to a request
message DummyReply {}

message RequestType { required string req = 1; }

message ResponseType { required string repl = 1; }
// A simple service that takes a RequestType message as an input parameter
// and returns a ResponseType message as an output parameter.
service SearchService { rpc Search(RequestType) returns (ResponseType); }
